app/models/reservation.rb

```diff
@@ -25,7 +25,7 @@ class Reservation < ActiveRecord::Base
   validate :not_in_past, :available, :check_banned, on: :create
 
   # correctly update the overdue flag if necessary
-  before_save :update_overdue, if: :checked_out?
+  before_save :update_overdue, if: :checked_out?, unless: self.changes.empty?
```

Comment:
@orenyk I'm not sure if adding another condition is an "and" or an "or". Also this doesn't compile form some reason (the rails s command doesn't run).

Any tips?